src/app/Library/CrudPanel/Traits/Create.php

```diff
@@ -264,11 +264,15 @@ private function handleManyRelationItemRemoval($modelInstance, $removedEntries,
             return $removedEntries->delete();
         }
 
-        if (! $relationColumnIsNullable && $modelInstance->dbColumnHasDefault($relationForeignKey)) {
-            return $removedEntries->update([$relationForeignKey => $modelInstance->getDbColumnDefault($relationForeignKey)]);
+        $dbColumnDefault = $modelInstance->getDbColumnDefault($relationForeignKey);
+
+        if ($relationColumnIsNullable) {
+            return $removedEntries->update([$relationForeignKey => $dbColumnDefault]);
+        } else {
+            return ! is_null($dbColumnDefault) ?
+                $removedEntries->update([$relationForeignKey => $dbColumnDefault]) :
+                $removedEntries->delete();
```

Comment:
Looks like this would do the same thing but a little cleaner, right?

```suggestion
        if ($relationColumnIsNullable || $dbColumnDefault !== null) {
            return $removedEntries->update([$relationForeignKey => $dbColumnDefault]);
        } else {
            return $removedEntries->delete();
```

Reply:
Sure, thanks 👍